src/utils.c

```diff
@@ -283,10 +283,24 @@ ts_timestamp_bucket(PG_FUNCTION_ARGS)
 	Timestamp	timestamp = PG_GETARG_TIMESTAMP(1);
 	Timestamp	result;
 	int64		period = -1;
+	/* The offset moves the epoch to start on a monday the default postgres epoch starts on a saturday.
+	 * This makes time-buckets by a week more intuitive.
+	 */
+#ifdef HAVE_INT64_TIMESTAMP
+	int64 offset = 2*USECS_PER_DAY;
```

Comment:
Why aren't we passing in the user-given offset to the function here? It seems strange that we handle offsets in SQL but also here. It would make more sense to deal with all offsets in one place.

It would also fix the slightly weird EXPLAIN output with offsets where the offset is printed twice and doesn't correspond to the user's input:

```
time_bucket('@ 1 min'::interval, (hyper_1."time" - '@ 30 secs'::interval)) + '@ 30 secs'::interval
```

Reply:
Can you also elaborate on the difference between what we are doing here (just adding 2 days) vs what is done in `date_trunc` by using `date2isoweek()`? This seems very simple compared to what's done their and I am wondering if we're missing something?

Reply:
Yes so in general date_trunc tries to align with human notions of weeks, years, etc. We don't. time bucket is a straight truncation of the time using a constant-size interval from the epoch. We have never tried to align our truncation to human notions (which is why we don't support months or years). for time-bucket '1 week' is really same as '7 days'.

Reply:
But aren't we trying to align the human-notion of a week now? Wasn't that the whole point with this PR? So what is the difference really?

Also, surprised to learn that we do not support month or years. I understand that the original implementation didn't align our truncation to human notions but isn't that what we found was the right thing to do now?

It is not clear to me why we simply cannot call `date_trunc` underneath the hood for the intervals that are supported by that function. Then we'd surely be compatible. I guess I am missing something?